pkg/portworx/pxVolumeOps.go

```diff
@@ -148,8 +149,23 @@ func (p *pxVolumeOps) GetVolumes() ([]*api.SdkVolumeInspectResponse, error) {
 		}
 	} else {
 		// If it is no volumes (all)
-		volsInfo, err := volumes.InspectWithFilters(p.pxVolumeOpsInfo.Ctx,
-			&api.SdkVolumeInspectWithFiltersRequest{})
+		var (
+			err      error
+			volsInfo *api.SdkVolumeInspectWithFiltersResponse
+		)
+
+		// if label is set, use them as filter
+		if len(p.GetPxVolumeOpsInfo().Labels) == 0 {
```

Comment:
You can just pass the Labels to InspectWithFilters. It should do the right thing. No need for an `if` statement here.

Reply:
Fixed it.